python/ray/tests/test_autoscaler.py

```diff
@@ -683,6 +737,102 @@ def testAggressiveAutoscaling(self):
                 "CPU": 1
             }
 
+    def testRequestResourcesAfterIdleTimeout(self):
```

Comment:
Besides testing the idle timeout, can we add test cases for race conditions around resource demands?

Reply:
This already tests some of them, it only tests the IDLE timeout at the end, but tests different cases too.
Will add one more.